`KDTreeSingleIndexDynamicAdaptor_::operator=()` definition of implicit copy constructor is deprecated because it has a user-declared copy assignment operator

In modern C++ compilers such as clang-12 we also get this compile warning:

```
[...]/nanoflann.hpp:1756:38: warning: definition of implicit copy constructor for 'KDTreeSingleIndexDynamicAdaptor_<nanoflann::L2_Simple_Adaptor<double, ValueCache<int>::PointCloud>, ValueCache<int>::PointCloud, -1>' is deprecated because it has a user-declared copy assignment operator [-Wdeprecated-copy]
    KDTreeSingleIndexDynamicAdaptor_ operator=(
```

with the appropriate flag enabled (`-Wdeprecated-copy`). The warning can be silenced with 

```
    /** Explicitly default the copy constructor */
    KDTreeSingleIndexDynamicAdaptor_(
        const KDTreeSingleIndexDynamicAdaptor_& rhs) = default;
```

--- a/include/nanoflann.hpp
+++ b/include/nanoflann.hpp
@@ -1752,6 +1752,10 @@ class KDTreeSingleIndexDynamicAdaptor_
         BaseClassRef::m_leaf_max_size = params.leaf_max_size;
     }
 
+    /** Explicitly default the copy constructor */
+    KDTreeSingleIndexDynamicAdaptor_(
+        const KDTreeSingleIndexDynamicAdaptor_& rhs) = default;
+
     /** Assignment operator definiton */
     KDTreeSingleIndexDynamicAdaptor_ operator=(
         const KDTreeSingleIndexDynamicAdaptor_& rhs)

